### PR TITLE
Update 03-security-principles.md

### PR DESCRIPTION
**Pull request by sydseter**
Add more general principles on security architecture and design.

**Summary** :  

I was thinking that it would be a good idea to also have more general security principles here as well as they also can help give practical guidance on the implementation and design of an application/system.
There are already several principles here already that leans more to the architecture and design side (e.g: "psychological acceptability"), and it's not like these not are relevant to the concrete coding either. Specifically, they get mentioned in ISMS 27002:2022 under "Secure system architecture and engineering principles" where of the purpose of the control should be directly related to these "to ensure that information systems are securely **designed**, **implemented** and operated within the **development life cycle**". 

**Description for the changelog** :  

Add more general principles on security architecture and design.

**Other info** :  

  * [ISMS 27002:2022, 8.27: "Secure system architecture and engineering principles"](https://www.isms.online/iso-27001/annex-a/8-27-secure-system-architecture-engineering-principles-2022/)
  * [Usability and Manageability](https://pubs.opengroup.org/security/o-esa/#_Toc291061714)
  * [Security by Design](https://pubs.opengroup.org/security/o-esa/#_Toc291061712)
  * [Secure by default](https://en.wikipedia.org/wiki/Secure_by_default)
  * [Least Functionality](https://csf.tools/reference/nist-sp-800-53/r5/cm/cm-7/)

### CODE REVIEW

**Review by jgadsden — DISMISSED** [review]
thanks @sydseter , did you want to add your references to 'Other' at the bottom of the page? No problem either way, your decision
[ISMS 27002:2022, 8.27: "Secure system architecture and engineering principles"](https://www.isms.online/iso-27001/annex-a/8-27-secure-system-architecture-engineering-principles-2022/)
[Usability and Manageability](https://pubs.opengroup.org/security/o-esa/#_Toc291061714)
[Security by Design](https://pubs.opengroup.org/security/o-esa/#_Toc291061712)
[Secure by default](https://en.wikipedia.org/wiki/Secure_by_default)
[Least Functionality](https://csf.tools/reference/nist-sp-800-53/r5/cm/cm-7/)

**Comment by sydseter** [comment]
Perhaps to many references would make the reference list less clear. I decided to leave out some of the references and only keep the references directly related to security principles because of that.

I removed the SecDevOps referenced and use secure operational management instead. Reason being that it's more correct in relation to what I think these principles actually mean.

I'll add a reference to SAMM about Operational Management in case someone wonder what it means. https://owaspsamm.org/model/operations/operational-management/

**Comment by sydseter** [comment]
That should do it from my side.

File: draft/04-foundations/03-security-principles.md
Comment (jgadsden) [inline draft/04-foundations/03-security-principles.md:1]: you are welcome @sydseter to add your name to the contributors on line 6 if you like

Comment (sydseter) [reply draft/04-foundations/03-security-principles.md:1]: Thanks, hope I found the right place?

**Comment by sydseter** [comment]
@jgadsden my name is aparently mispelled?

**Review by jgadsden — APPROVED** [review]
thanks again for the contributions @sydseter , please go ahead and merge

**Pull request merged**